src/gui/editors/PianoRoll.cpp

```diff
@@ -1310,16 +1312,34 @@ void PianoRoll::keyPressEvent(QKeyEvent* ke)
 				int direction = (ke->key() == Qt::Key_Up ? +1 : -1);
 				if( ( ke->modifiers() & Qt::ControlModifier ) && m_action == ActionNone )
 				{
-					// shift selection up an octave
+					// shift selection by one octave
 					// if nothing selected, shift _everything_
 					if (hasValidMidiClip())
 					{
-						shiftSemiTone( 12 * direction );
+						// An octave could potentially be greater or less than twelve semitones if the microtuner is in use.
+						if (m_midiClip->instrumentTrack()->microtuner()->enabled())
+						{
+							int keymapSize = Engine::getSong()->getKeymap(m_midiClip->instrumentTrack()->microtuner()->currentKeymap())->getSize();
+							if (keymapSize > 0)
+							{
+								shiftSemiTone(keymapSize * direction);
+							}
+							else
+							{
+								// Determine octave size from the scale if the keymap isn't in use.
+								int scaleSize = Engine::getSong()->getScale(m_midiClip->instrumentTrack()->microtuner()->currentScale())->getIntervals().size() - 1;
+								shiftSemiTone(scaleSize * direction);
```

Comment:
```suggestion
						auto microtuner = m_midiclip->instrumentTrack()->microtuner();
						if (microtuner->enabled())
						{
							int keymapSize = Engine::getSong()->getKeymap(microtuner->currentKeymap())->getSize();
							if (keymapSize > 0)
							{
								shiftSemiTone(keymapSize * direction);
							}
							else
							{
								// Determine octave size from the scale if the keymap isn't in use.
								int scaleSize = Engine::getSong()->getScale(microtuner->currentScale())->getIntervals().size() - 1;
								shiftSemiTone(scaleSize * direction);
```
Let's shorten some of these lines just a little bit. :)